jobs/superkey.go

```diff
@@ -0,0 +1,105 @@
+package jobs
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/RedHatInsights/sources-api-go/dao"
+	l "github.com/RedHatInsights/sources-api-go/logger"
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/service"
+	"github.com/RedHatInsights/sources-api-go/util"
+)
+
+type SuperkeyDestroyJob struct {
+	Identity string
+	Tenant   int64
+	Model    string
+	Id       int64
+}
+
+func (sk SuperkeyDestroyJob) Delay() time.Duration {
+	// run this job immediately, no delay.
+	return 0
+}
+
+func (sk SuperkeyDestroyJob) Arguments() map[string]interface{} {
+	return map[string]interface{}{
+		"model": sk.Model,
+		"id":    sk.Id,
+	}
+}
+
+func (sk SuperkeyDestroyJob) Name() string {
+	return "SuperkeyDestroyJob"
+}
+
+func (sk SuperkeyDestroyJob) Run() error {
+	l.Log.Infof("Running [%v] with [%v]", sk.Name(), sk.Arguments())
+
+	switch sk.Model {
+	case "source":
+		return sk.sendForSource(sk.Id)
+	case "application":
+		return sk.sendForApplication(sk.Id)
+	default:
+		return fmt.Errorf("unsupported model for superkey: %v", sk.Model)
+	}
+}
+
+// Lists all applications for a source, sends the destroy request for each, then
+// enqueues deletion of itself.
+
+// the sub-jobs enqueue deletion of their respective resources.
+func (sk SuperkeyDestroyJob) sendForSource(id int64) error {
+	l.Log.Infof("Sending SuperKey Delete request for source %v", sk.Id)
+
+	a := dao.GetApplicationDao(&sk.Tenant)
+
+	apps, _, err := a.SubCollectionList(&m.Source{ID: id}, 100, 0, make([]util.Filter, 0))
```

Comment:
I just want to double check that 100 is a reasonable limit for this. I am expecting a "yes, it is" response, but I wanted to ask nonetheless :)

Reply:
Yep, most of our sources only have 1 application, the sub-applications (cost, swatch) only support one connection per credential. SO realistically the max is 2.

Reply:
Cool, thank you!